x-pack/plugins/profiling/server/lib/setup/fleet_policies.ts

```diff
@@ -250,6 +256,11 @@ export async function createSymbolizerPackagePolicy({
   packagePolicyClient,
   config,
 }: ProfilingSetupOptions) {
+  const symbolizerPolicy = await getSymbolizerPolicy({ packagePolicyClient, soClient });
+  // In case symbolizer policy has been already created do not try to create it again
+  if (symbolizerPolicy) {
+    return;
+  }
```

Comment:
I have an alternative suggestion that solves the same issue above but for all functions responsible for setup. It has the benefit that the validation steps are only run once and gets us closer to making the endpoint idempotent. It also keeps the logic inside the `setup` route instead of moving it into the respective setup functions.

For instance, replace the following block in `x-pack/plugins/profiling/server/routes/setup.ts with the second block:

```
        const executeFunctions = [
          installLatestApmPackage,
          updateApmPolicy,
          createCollectorPackagePolicy,
          createSymbolizerPackagePolicy,
          enableResourceManagement,
          setSecurityRole,
          setMaximumBuckets,
        ];
```

```
        const executeFunctions = [
          ...(mergedState.packages.installed ? [] : [installLatestApmPackage]),
          ...(mergedState.policies.apm.installed ? [] : [updateApmPolicy]),
          ...(mergedState.policies.collector.installed ? [] : [createCollectorPackagePolicy]),
          ...(mergedState.policies.symbolizer.installed ? [] : [createSymbolizerPackagePolicy]),
          ...(mergedState.resource_management.enabled ? [] : [enableResourceManagement]),
          ...(mergedState.permissions.configured ? [] : [setSecurityRole]),
          ...(mergedState.settings.configured ? [] : [setMaximumBuckets]),
        ];
```

I made use of the `mergedState` data structure that contains the results of the validation functions. If a particular field in `mergedState` is false, then we add the corresponding setup function to the list of executed functions. Otherwise, we skip.